### PR TITLE
imp: replace `Sender/ReceiverChainIsSource` with `HasPrefix`

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

ref: #6696

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/relay.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func (k Keeper) sendTransfer(
 		// chain inside the packet data. The receiving chain will perform denom
 		// prefixing as necessary.
 
-		if token.Denom.SenderChainIsSource(sourcePort, sourceChannel) {
+		if !token.Denom.HasPrefix(sourcePort, sourceChannel) {
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:118]: I think if we do this change, we should flip flop the if statement. I think there's a bit of unnecessary mental gymnastics to figure out what a non prefix set means. I feel like it's easier to write a comment which states:
```go
// if the port and channel we are sending the token on was prefixed to this denom
// then we must be returning the token back to the chain they originated from
```
then I think the burn coins logic is more natural

File: modules/apps/transfer/keeper/relay.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		// NOTE: We use SourcePort and SourceChannel here, because the counterparty
 		// chain would have prefixed with DestPort and DestChannel when originally
 		// receiving this coin as seen in the "sender chain is the source" condition.
-		if token.Denom.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel()) {
+		if token.Denom.HasPrefix(packet.GetSourcePort(), packet.GetSourceChannel()) {
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:204]: I personally don't feel like this reads easier, but I'm okay with the change. I think either restructing the comments or flip flopping the if statements could make it better, but as is requires me to reparse all the logic to figure out which chain is source and which port and channel to be passing in

Comment (colin-axner) [reply modules/apps/transfer/keeper/relay.go:204]: maybe you could update the comment:
```go
// if the token we are receiving has the packet source port and channel prefixed
// then the sending chain must have prefixed this hop to the token when it originally
// received these tokens from us. Given that we are receiving back the tokens
// we should remove the prefix added by the counterparty and unescrow the tokens. 
```

I think the previous comment is more relevant to the previous function and even references it ""sender chain is source" condition"

File: modules/apps/transfer/keeper/relay.go [hunk 3]
```diff
@@ -347,7 +347,7 @@ func (k Keeper) refundPacketTokens(ctx sdk.Context, packet channeltypes.Packet,
 			return err
 		}
 
-		if token.Denom.SenderChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel()) {
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:350]: I have a feeling that if we are changing to `HasPrefix`, the logic will always be easier to parse if the first if statement is not negated

**Comment by crodriguezvega** [comment]
@colin-axner I think I have addressed your comments. Please take a look when you have time. I agree it makes sense to not negate the condition. :)

File: modules/apps/transfer/keeper/relay.go [hunk 4]
```diff
@@ -356,29 +357,29 @@ func (k Keeper) refundPacketTokens(ctx sdk.Context, packet channeltypes.Packet,
...
+		// if the token we must refund is prefixed by the source port and channel
+		// then the tokens were burnt on the sending chain and we must mint new tokens
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:361]:
```suggestion
		// then the tokens were burnt when the packet was sent and we must mint new tokens
```

**Review by colin-axner — APPROVED** [review]
Thank you for updating the if else, I think it's more natural to reason about

**Review by chatton — APPROVED** [review]
LGTM! This reads a good bit better now i think, thanks for the cleanup :muscle:

**Pull request merged**